### ISSUE TITLE
How to fine-tune the MPB model on our binary classification dataset following the approach described in the paper?

### ISSUE THREAD

**Issue by YuxuanLou** [Thurs-Lw/MPB-EXP-MUT#7]
Thank you for your outstanding work! Could you please guide us on how to fine-tune the MPB model on our binary classification dataset following the approach described in the paper? Would you be able to update the README with detailed instructions regarding this implementation? We sincerely appreciate your assistance!
感谢您出色的工作！请问如何像文章里那样使用MPB模型在自己的二分类数据集上微调？您能在readme里更新有关这方面的指导吗？再次向您表示感谢！

**Comment by Thurs-Lw** [comment]
If you want to fine-tune MP-TRANS on your own binary classification dataset, you can run the Python script using the following two commands:

#### 1. Convert your own dataset into MindRecord format:

```
python generate_dataset/generate_seq_for_classification.py
--data_dir <your dataset path>
--vocab_file model/vocab_v2.txt
--output_dir <MindRecord format dataset output path>
 --max_seq_length 1024
--do_train True
--do_eval True
--do_test True
```
⚠️You need to ensure that the **data_dir** path contains the files **"train.csv", "val.csv", and "test.csv"**. The columns in these files should be comma-separated, and the column names must include **id**, **seq**, and **label**, which correspond to the protein ID, protein sequence, and binary classification label, respectively.

such as:
> id,seq,label
> P12345,MKTVRQGVSIPFISLR,1
> P67890,GAVLILLLFVAAALA,0
> P54321,MVLSPADKTNVKAAW,1

#### 2. Fine-tuning training and evaluation script

```
python model/mpbert_classification.py
--config_path model/config_1024.yaml
 --do_train True 
--do_eval True 
--description classification 
--num_class 2 
--epoch_num 200 
--early_stopping_rounds 50 
--frozen_bert False 
--device_id 0 
--data_url <step 1 output_dir>
--load_checkpoint_url <download pretrain ckpt file from zenodo>
--output_url <output model path>
 --task_name <your task name>
```
After running the script, you should obtain a file named `<task_name>_Best_Model.ckpt`.

#### 3. Use the trained model to predict proteins

```
python model/mpbert_classification.py
--config_path model/config_1024.yaml
--do_predict True
--description classification 
--num_class 2 
--vocab_file model/vocab_v2.txt
--device_id 0 
--return_csv True
--data_url <your protein csv file>
--load_checkpoint_url <step 2 output ckpt file>
--output_url <output predict file path>
```
⚠️In this step, the `data_url` must point to a CSV file that contains the columns **id** and **seq**.

**Comment by YuxuanLou** [comment]
Thank you very much! Will the pre-trained parameters change? Does the fine-tuning process in the article alter the pre-trained parameters?
非常感谢！请问预训练的参数会改变吗?文章里的微调过程是否改变了预训练的参数？

**Comment by Thurs-Lw** [comment]
The code adopts a fine-tuning strategy that continues to train (i.e., does not freeze) the pre-trained parameters; therefore, the pre-trained weights and parameters will be updated.
